.github/workflows/vanilla_ubuntu.yml

```diff
@@ -0,0 +1,148 @@
+        run: cp .key testing/.key
+
+      - name: Terraform Validate
+        id: validate
+        run: |
+          cd testing/
+          terraform validate -no-color
+
+      - name: Terraform Plan
+        id: plan
+        run: |
+          cd testing/
+          terraform plan -var "GITHUB_RUN_ID=$GITHUB_RUN_ID" -no-color
+        continue-on-error: true
+
+      - name: Terraform Plan Status
+        if: steps.plan.outcome == 'failure'
+        run: exit 1
+
+      - name: Terraform Apply
+        run: |
+          cd testing/
+          terraform apply  -var "GITHUB_RUN_ID=$GITHUB_RUN_ID" -auto-approve
+
+      - name: Set up Python
+        id: setup_python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip3 install ansible pytest-testinfra
+
+      - name: Ansible Version check
+        run: ansible --version
+
+      - name: Create inventory hosts.ini
+        run: |
+          touch hosts.ini
+          echo "[rke2_servers]" > hosts.ini
+          aws ec2 describe-instances --filters "Name=tag:Owner,Values=rke2-ansible-github-actions" "Name=tag:NodeType,Values=Server" "Name=tag:github_run,Values=$GITHUB_RUN_ID" --query "Reservations[*].Instances[*].PublicIpAddress" --output text >> hosts.ini
+          echo "[rke2_agents]" >> hosts.ini
+          aws ec2 describe-instances --filters "Name=tag:Owner,Values=rke2-ansible-github-actions" "Name=tag:NodeType,Values=Agent" "Name=tag:github_run,Values=$GITHUB_RUN_ID" --query "Reservations[*].Instances[*].PublicIpAddress" --output text >> hosts.ini
+          echo "" >> ansible.cfg
+          echo "" >> ansible.cfg
+          echo "remote_user=ubuntu" >> ansible.cfg
+          echo "private_key_file=.key" >> ansible.cfg
+
+      - name: Check hosts.ini and ansible.cfg
+        run: |
+          cat hosts.ini
+          cat ansible.cfg
+
+      - name: Prep inventory
+        run: |
+          cp -R inventory/sample inventory/ubuntu20
+          cp hosts.ini inventory/ubuntu20/hosts.ini
+
+      - name: Run playbook
+        run: |
+          ANSIBLE_HOST_KEY_CHECKING=False ansible-playbook -i inventory/ubuntu20/hosts.ini -u ubuntu --verbose --private-key .key site.yml
+
+      - name: Run Ansible Tests
+        run: |
+          ANSIBLE_HOST_KEY_CHECKING=False ansible-playbook -i inventory/ubuntu20/hosts.ini -u ubuntu --verbose --private-key .key testing.yml
+
+      - name: Run Python Tests
+        run: |
+          export DEFAULT_PRIVATE_KEY_FILE=.key
+          pytest --hosts=rke2_servers --ansible-inventory=inventory/ubuntu20/hosts.ini --force-ansible --connection=ansible --sudo testing/basic_server_tests.py
+          pytest --hosts=rke2_agents --ansible-inventory=inventory/ubuntu20/hosts.ini --force-ansible --connection=ansible --sudo testing/basic_agent_tests.py
+
+      - name: Delete Stack
+        if: ${{ always() }}
```

Comment:
@dgvigil  - Is it possible to throw in a conditional scenario for debugging use cases...i.e., if playbook fails at certain step AND `DEBUG_ENV=true` then keep environment around so that I can login and debug?

Reply:
again, possibly a future issue than can be worked on, not critical

Reply:
For local dev, I would suggest running the Terraform commands locally or create a Vagrantfile to test locally.
ie `AWS_ACCESS_KEY_ID="key id" AWS_SECRET_ACCESS_KEY="your key" terraform apply -var "GITHUB_RUN_ID=999"`

Reply:
updating comment to include reference to issue #78 .  Based on our conversation on issues with how to do destructive development testing and iterative testing - having the ability to create local, transient environments on-demand as a means of doing quick debugging (without hacking up the repo with a ton of customization files) - I'm proposing that we do include molecule folder to do the end-to-end testing - whether it's local or remote against other providers...